code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -184,3 +184,98 @@
 	..()
 	_color = null
 	update_icon()
+
+/obj/item/weapon/melee/energy/hfmachete
+	name = "high-frequency machete"
+	desc = "A high-frequency broad blade used either as an implement or in combat like a short sword."
+	icon_state = "hfmachete0"
+	var/base_state = "hfmachete"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/swords_axes.dmi', "right_hand" = 'icons/mob/in-hand/right/swords_axes.dmi')
+	sharpness_flags = SHARP_BLADE | SERRATED_BLADE | CHOPWOOD
+	force = 13 // You can be crueler than that, Jack.
+	throwforce = 20
+	throw_speed = 8
+	throw_range = 8
+	w_class = W_CLASS_MEDIUM
+	flags = FPRINT
+	siemens_coefficient = 1
+	origin_tech = Tc_COMBAT + "=3" + Tc_SYNDICATE + "=3"
+	attack_verb = list("attacks", "dices", "cleaves", "tears", "cuts", "slashes",)
+
+/obj/item/weapon/melee/energy/hfmachete/update_icon()
+	icon_state = "[base_state][active]"
+
+/obj/item/weapon/melee/energy/hfmachete/attack_self(mob/living/user)
+	toggleActive(user)
+	add_fingerprint(user)
+
+/obj/item/weapon/melee/energy/hfmachete/proc/toggleActive(mob/user, var/togglestate = "")
+	switch(togglestate)
+		if("on")
+			active = 1
+		if("off")
+			active = 0
+		else
+			active = !active
+	if(active)
+		force = 25
+		throwforce = 6
+		throw_speed = 3
+		sharpness = 1.7
+		sharpness_flags += HOT_EDGE
+		to_chat(user, "<span class='warning'> [src] starts vibrating.</span>")
+		playsound(user, 'sound/weapons/hfmachete1.ogg', 40, 0)
+	else
+		force = initial(force)
+		throwforce = initial(throwforce)
+		throw_speed = initial(throw_speed)
+		sharpness = initial(sharpness)
+		sharpness_flags = initial(sharpness_flags)
+		to_chat(user, "<span class='notice'> [src] stops vibrating.</span>")
+		playsound(user, 'sound/weapons/hfmachete0.ogg', 40, 0)
+	update_icon()
+
+/obj/item/weapon/melee/energy/hfmachete/throw_at(atom/target, range, speed, override = 1)
+	if(!usr)
+		return ..()
+	spawn()
+		playsound(src, get_sfx("machete_throw"),30, 0)
+		animate(src, transform = turn(matrix(), -30), time = 1, loop = -1)
+		animate(transform = turn(matrix(), -60), time = 1)
+		animate(transform = turn(matrix(), -90), time = 1)
+		animate(transform = turn(matrix(), -120), time = 1)
+		animate(transform = turn(matrix(), -150), time = 1)
+		animate(transform = null, time = 1)
+		while(throwing)
+			sleep(5)
+		animate(src)
+	..(target, range, speed, override, fly_speed = 3)
+
+/obj/item/weapon/melee/energy/hfmachete/throw_impact(atom/hit_atom)
+	if(isturf(hit_atom))
+		for(var/mob/M in hit_atom)
+			playsound(M, get_sfx("machete_throw_hit"),60, 0)
+	..()
+
+/obj/item/weapon/melee/energy/hfmachete/attack(target as mob, mob/living/user as mob)
+	if(active && istype(target, /obj/effect/plantsegment)) // really useful in a jungle..
+		var/obj/effect/plantsegment/P = istype(target, /obj/effect/plantsegment)
```

Comment:
heck at least do `for plantsegment/P in range(1)` to make it more useful than a welder